Documentation/learning/why.md

```diff
@@ -71,12 +71,30 @@ If an application reasons primarily about metadata or metadata ordering, such as
 
 In theory, it’s possible to build these primitives atop any storage systems providing strong consistency. However, the algorithms tend to be subtle; it is easy to develop a locking algorithm that appears to work, only to suddenly break due to  thundering herd and timing skew. Furthermore, other primitives supported by etcd, such as transactional memory depend on etcd’s MVCC data model; simple strong consistency is not enough.
 
 For distributed coordination, choosing etcd can help prevent operational headaches and save engineering effort.
 
+### Notes on the usage of lock and lease
+etcd provides [lock APIs][etcd-v3lock] which is based on [the lease mechanism][lease] and [its implementation in etcd][etcdlease]. The basic idea of the lease mechanism is like this: a server grants a token, which is called lease, to a requesting client. When the server grants a lease, it associates TTL with the lease. When the server detects passing of time longer than TTL, it revokes the lease. During the client holds a non revoked lease, it can claim that it owns an access to a resource associated with the lease. In the case of etcd, the resource is a key. etcd provides lock APIs with this scheme.
+
+The most important aspect of the lease mechanism is that TTL is defined as a physical time interval. Both of the server and client measures passing of time with their own clocks, which are not synchronized each other. It allows a situation that the server revokes the lease but the client still claims it owns the lease because of clock drift.
+
+Then how the lease mechanism guarantees mutual exclusion of the locking mechanism? Actually, the lease mechanism itself doesn't guarantee mutual exclusion. Owning a lease cannot guarantee the owner holds a lock of the resource.
+
+In the case of etcd, mutual exclusion is implemented baesd on the mechanism of version number validation (it is sometimes called compare and swap in other systems like Consul). In etcd's RPCs like `Put` or `Txn`, we can specify required conditions about revision number and lease ID for the operations. If the conditions are not satisfied, the operation can be failed. With this mechanism, etcd provides distributed locking for clients. It means that a client knows that it is acquiring a lock of a key when its requests are completed by etcd cluster successfully.
+
+We can find almost same ideas and designs of distributed locking mechanisms in the below literature:
```

Comment:
"We can find almost same ideas and designs of distributed locking mechanisms in the below literature:" -> "In distributed locking literature similar designs are described:"